### PR TITLE
Rsp2 and bug fix

### PR DESCRIPTION
**Pull request by grburgess**
A few cleanups in EventList and RSP2 handling

### CODE REVIEW

File: threeML/plugins/EventListLike.py [hunk 1]
```diff
@@ -73,11 +71,55 @@ def __init__(self, name, event_list, background_selections, source_intervals, rs
...
+        try:
+
+            OGIPLike.__init__(self,
+                              name,
+                              pha_file=self._observed_pha,
+                              bak_file=self._bkg_pha,
+                              rsp_file=rsp_file,
+                              verbose=verbose)
+
+        except(NegativeBackground):
```
Comment (giacomov) [inline threeML/plugins/EventListLike.py]: If I understand well what is going on here, I am not sure this is right.

A negative background does not necessarily mean that the polynomial grade is too high... I'd rather throw an exception and advise to change the background off-pulse intervals.

File: threeML/plugins/OGIP/eventlist.py [hunk 2]
```diff
@@ -1062,12 +1113,10 @@ def __init__(self, arrival_times, energies, n_channels, start_time=None, stop_ti
 
             self._dead_time = None
 
-
-
-    def _exposure_over_interval(self, tmin, tmax):
+    def exposure_over_interval(self, tmin, tmax):
```
Comment (giacomov) [inline threeML/plugins/OGIP/eventlist.py]: Is this method supposed to be called by the user?

If yes, then you should add a detailed docstring.

If not, then you should call it _exposure_over_interval (also in the parent class).

File: threeML/plugins/EventListLike.py [hunk 3]
```diff
@@ -1,10 +1,8 @@
 __author__ = 'grburgess'
 
 import numpy as np
-import matplotlib.pyplot as plt
-import pandas as pd
-import warnings
-import re
+
+from threeML.exceptions.custom_exceptions import custom_warnings, NegativeBackground
```
Comment (grburgess) [inline threeML/plugins/EventListLike.py]: Aha, ok. I was trying to keep them central. I'll stop that!

File: threeML/plugins/EventListLike.py (hunk 1 above)
Comment (grburgess) [inline threeML/plugins/EventListLike.py]: Fixed.

File: threeML/plugins/OGIP/eventlist.py (hunk 2 above)
Comment (grburgess) [inline threeML/plugins/OGIP/eventlist.py]: It is not supposed to be called by the user, but it is called from an EventList object externally. Therefore, I believe that it should be public. (unless I misinterpret the public/private concept in python) I have added a detailed doctstring.

Note, EventList is not exposed to the user.

**Pull request merged**